File: 0392. Is Subsequence/392_naive.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        sct, tct, slen, tlen = 0, 0, len(s), len(t)
        while sct < slen and tct < tlen:
            if s[sct] == t[tct]:
                sct += 1

            tct += 1

        return sct == slen

File: 0392. Is Subsequence/392_naive.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        size_t ct_s = 0, ct_t = 0;
        size_t len_s = s.length(), len_t = t.length();
        while(ct_s < len_s and ct_t < len_t){
            if(s.at(ct_s) != t.at(ct_t)){
                ct_t++;
            }
            else{
                ct_s++;
                ct_t++;
            }
        }
        return true ? ct_s == len_s : false;
    }
};
